### PR TITLE
Face Recognition Using Wavelet Features

### LINKED ISSUE #5889: Face recognition using wavelet features in Matlab
### Topic Suggestion
Face recognition using wavelet features in Matlab

#### Introduction
A face recognition system is a technology capable of matching human faces from digital images or videos frame against face database. Wavelet is a wave-like oscillation with an amplitude that begins at zero, increases or decreases, and returns to zero. These wavelet coefficients are used to extract features from hyperspectral data. These extracted features are called wavelet features.
In this tutorial, a face detection scheme is implemented using the wavelet features. We use wavelet features to extract facial features and use principal component analysis to reduce the wavelet feature vectors. The proposed scheme is very robust and capable of recognizing the faces even if some changes occur in the face, such as growing beards, mustache e.t.c

#### Key takeaways 
- Face image database preparation for training and testing.
- Finding wavelet features.
- PCA for dimension reduction.
- Training and testing procedures.

#### Article quality
This article combines the idea of extracting wavelet feature and using a different algorithm PCA to reduce the dimensions before training and testing.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by simonmwaniki**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5889 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/face-recognition-using-wavelet-features/index.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+Where;
+
+- `fvstd`- Is the input feature vector, which in our case is the vector of size 380.
+- `M` - Is the mean of all the feature vectors.
+- `Ppca` - Is the transformation matrix.
+- `fvpca` - Is the reduced feature vector.
+
+### Training
+It involves getting the wavelet feature and projecting them to the PCA space.
```
Comment (Dawe-7) [inline content/articles/face-recognition-using-wavelet-features/index.md:75]:
```suggestion
It involves getting the wavelet features and projecting them to the PCA space.
```

File: content/articles/face-recognition-using-wavelet-features/index.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+eigenvalues
+
+![Distorted image](/engineering-education/facrecognition-using-wavelet-features/wavelet-five.png)
+
+Distorting some image parts
+
+![Distored image](/engineering-education/facrecognition-using-wavelet-features/wavelet-six.png)
+
+### Conclusion
+The wavelet feature can be used to make a face recognition system. Using this method, we find a very robust method. Even if the image is distorted, it is still possible to recognize the faces accurately using these features.
```
Comment (Dawe-7) [inline content/articles/face-recognition-using-wavelet-features/index.md:291]:
```suggestion
In this tutorial, we have looked at how the wavelet feature is used to make a face recognition system. This method as we have seen is a very robust method. Even if the image is distorted, it is still possible to recognize the faces accurately using these features.

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Nice tutorial @simonmwaniki, I've left you a few suggestions, kindly make the changes and we'll move on to tech review.

**Comment by simonmwaniki** [comment]
@Dawe-7 changes made.

**Review by Dawe-7 — COMMENTED** [review]
Hello @simonmwaniki, as i was doing a final review on my end before marking this article as `ready for final review` i noticed that you have extra images. I haven't seen any image links to them in the article, if they are unnecessary kindly remove them. There also two hero images, kindly check on that. 

Ping me when done.

**Comment by Dawe-7** [comment]
> Hello @simonmwaniki, as i was doing a final review on my end before marking this article as `ready for final review` i noticed that you have extra images. I haven't seen any image links to them in the article, if they are unnecessary kindly remove them. There also two hero images, kindly check on that.
> 
> Ping me when done.

Any updates @simonmwaniki

**Comment by simonmwaniki** [comment]
@Dawe-7 I checked the images, they were ok and all links. No extra image. Check slack please.

**Review by Dawe-7 — APPROVED** [review]
Awesome Job on the article @simonmwaniki. The unwanted images were outside your article but pushed together with it and i guess that's why you missed it. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles. Thank you.

**Comment by ahmadmardeni1** [comment]
Great work team. @Dawe-7 @simonmwaniki

**Pull request merged**